src/java.base/share/classes/sun/security/ssl/TransportContext.java

```diff
@@ -458,6 +455,12 @@ SSLException fatal(Alert alert, String diagnostic,
         }
     }
 
+    private void terminateHandshakeContext() {
+        handshakeCtxLock.lock();
+        handshakeContext = null;
+        handshakeCtxLock.unlock();
```

Comment:
The usual pattern is to use try { } finally { } with locks (even though in this particular case I don't see how the assignment would throw) - but if more things need to be done in the future here then at least the try { } finally { } will be in place.

```suggestion
        handshakeCtxLock.lock();
        try {
            handshakeContext = null;
        } finally {
            handshakeCtxLock.unlock();
        }    
```

Reply:
I added the try/finally.